src/control/server/ctl_storage_rpc_test.go

```diff
@@ -464,1208 +460,1070 @@ func TestServer_CtlSvc_StorageScan_PreEngineStart(t *testing.T) {
-
-			// In production, during server/server.go:srv.addEngines() and after
-			// srv.createEngine(), engine.storage.SetBdevCache() is called to load the
-			// results of the start-up bdev scan from the control service storage
-			// provider into the engine's storage provider. The control service and
-			// each of the engines have distinct storage provider instances so cached
-			// cached results have to be explicitly shared so results are available when
-			// engines are up.
-
-			for idx, ec := range engineCfgs {
-				var ebmbc *bdev.MockBackendConfig
-				if tc.eCtrlrs != nil && len(tc.eCtrlrs) > idx {
-					log.Debugf("bdevs %v to be returned for engine %d scan",
-						*tc.eCtrlrs[idx], idx)
-					ebmbc = &bdev.MockBackendConfig{
-						ScanRes: &storage.BdevScanResponse{
-							Controllers: *tc.eCtrlrs[idx],
-						},
-					}
-				}
-
-				// replace harness instance with mock I/O Engine
-				// to enable mocking of harness instance drpc channel
-				sp := storage.MockProvider(log, idx, &ec.Storage,
-					cs.storage.Sys, // share system provider cfo
-					scm.NewMockProvider(log, tc.smbc, nil),
-					bdev.NewMockProvider(log, ebmbc), nil)
-				if tc.eCtrlrs != nil && len(tc.eCtrlrs) > idx {
-					sp.SetBdevCache(storage.BdevScanResponse{
-						Controllers: *tc.eCtrlrs[idx],
-					})
-				}
-				te := newTestEngine(log, false, sp, ec)
-
-				if tc.enginesNotReady {
-					te.ready.SetFalse()
-				}
-
-				// mock drpc responses
-				dcc := new(mockDrpcClientConfig)
-				if tc.junkResp {
-					dcc.setSendMsgResponse(drpc.Status_SUCCESS,
-						makeBadBytes(42), nil)
-				} else if len(tc.drpcResps) > idx {
-					t.Logf("setting %d drpc responses for engine %d",
-						len(tc.drpcResps[idx]), idx)
-					dcc.setSendMsgResponseList(t, tc.drpcResps[idx]...)
-				} else {
-					t.Fatal("drpc response mocks unpopulated")
-				}
-				te.setDrpcClient(newMockDrpcClient(dcc))
-				te._superblock.Rank = ranklist.NewRankPtr(uint32(idx + 1))
-				for _, tc := range te.storage.GetBdevConfigs() {
-					tc.Bdev.DeviceRoles.OptionBits = storage.OptionBits(storage.BdevRoleAll)
-				}
-				md := te.storage.GetControlMetadata()
-				md.Path = "/foo"
-				md.DevicePath = md.Path
-
-				cs.harness.instances[idx] = te
-			}
-			cs.harness.started.SetTrue()
-
-			if tc.req == nil {
-				tc.req = &ctlpb.StorageScanReq{
-					Scm:  new(ctlpb.ScanScmReq),
-					Nvme: new(ctlpb.ScanNvmeReq),
-				}
-			}
-
-			if tc.scanTwice {
-				_, err := cs.StorageScan(test.Context(t), tc.req)
-				test.CmpErr(t, tc.expErr, err)
-				if err != nil {
-					return
-				}
-			}
-
-			resp, err := cs.StorageScan(test.Context(t), tc.req)
-			test.CmpErr(t, tc.expErr, err)
-			if err != nil {
-				return
-			}
-
-			if diff := cmp.Diff(tc.expResp, resp, defStorageScanCmpOpts...); diff != "" {
-				t.Fatalf("unexpected response (-want, +got):\n%s\n", diff)
-			}
-		})
-	}
-}
+//func TestServer_CtlSvc_StorageScan_PostEngineStart(t *testing.T) {
```

Comment:
Should be removed ?

Reply:
I will remove in the follow-up, I'm leaving commented to remind me to make sure everything is covered.